### PR TITLE
Added ability to change user password

### PR DESCRIPTION
**Pull request by azhavoro**
Added ability to change user password

### How has this been tested?
Manual testing

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat-ui/src/actions/auth-actions.ts [hunk 1]
```diff
@@ -100,3 +117,30 @@ export const authorizedAsync = (): ThunkAction => async (dispatch) => {
...
+            isReachable(`${cvat.config.backendAPI}/auth/password/change`, 'OPTIONS'),
+        ];
+        const [allowChangePassword] = await Promise.all(promises);
+
+        dispatch(authActions.loadServerAuthActionsSuccess(allowChangePassword));
+
+    } catch (error) {
+        dispatch(authActions.loadServerAuthActionsFailed(error));
+    }
+}
```
Comment (bsekachev) [inline cvat-ui/src/actions/auth-actions.ts:146]: Why do we need check if server allows to change password? Can it be false?

File: cvat/settings/base.py [hunk 2]
```diff
@@ -235,6 +235,7 @@ def generate_ssh_keys():
 
 # https://github.com/pennersr/django-allauth
 ACCOUNT_EMAIL_VERIFICATION = 'none'
+OLD_PASSWORD_FIELD_ENABLED = True
```
Comment (bsekachev) [inline cvat/settings/base.py:238]: Is this option used somehow?

File: cvat-ui/src/reducers/interfaces.ts [hunk 3]
```diff
@@ -13,6 +13,10 @@ export interface AuthState {
     initialized: boolean;
     fetching: boolean;
     user: any;
+    authActionsFetching: boolean;
+    authActionsInitialized: boolean,
```
Comment (bsekachev) [inline cvat-ui/src/reducers/interfaces.ts:17]: ;

File: cvat-ui/src/containers/change-password-dialog/change-password-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        onChangePassword(oldPassword: string, newPassword1: string, newPassword2: string): void {
+            dispatch(changePasswordAsync(oldPassword, newPassword1, newPassword2));
+        },
+    });
+}
+
+export default connect(
+    mapStateToProps,
+    mapDispatchToProps,
+)(ChangePasswordModalComponent);
```
Comment (bsekachev) [inline cvat-ui/src/containers/change-password-dialog/change-password-dialog.tsx:40]: I believe we can move it to ``components/change-password-modal/change-password-modal`` and remove this file. No need to have separated component file to getting state from redux storage.

File: cvat-ui/src/components/change-password-modal/change-password-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                                changePasswordData.newPassword2,
+                            );
+                        }}
+                        fetching={fetching}
+                    />
+            </Modal>
+        );
+}
+
+export default withRouter(ChangePasswordComponent);
```
Comment (bsekachev) [inline cvat-ui/src/components/change-password-modal/change-password-modal.tsx:52]: I don't see how you use RouterProps here

**Comment by bsekachev** [comment]
@ActiveChooN 

Could you please also take a look at this PR?

**Comment by bsekachev** [comment]
@azhavoro Please, update minor versions for ``cvat-ui`` and ``cvat-core``

File: cvat-ui/src/actions/auth-actions.ts (hunk 1 above)
Comment (azhavoro) [reply cvat-ui/src/actions/auth-actions.ts:146]: Yes, if LDAP is used for example. This API endpoint is available only for basic authorization.

File: cvat/settings/base.py (hunk 2 above)
Comment (azhavoro) [reply cvat/settings/base.py:238]: This option requires current user password for verification and used by django-rest-auth.

File: cvat-ui/src/containers/change-password-dialog/change-password-dialog.tsx (hunk 4 above)
Comment (azhavoro) [reply cvat-ui/src/containers/change-password-dialog/change-password-dialog.tsx:40]: Got it

**Comment by ActiveChooN** [comment]
@azhavoro, can you also fix linting problems related to your PR, please?

**Comment by nmanovic** [comment]
@zhiltsov-max , could you please look at the PR and test it?

**Pull request merged**